File: LeetCode/blind75/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        longestSequence = 0

        for num in numSet:
            parentSequence = num - 1
            if parentSequence not in numSet:
                length = 1

                while (num + length) in numSet:
                    length += 1

                longestSequence = max(length, longestSequence)

        return longestSequence
        

File: LeetCode/blind75/128. Longest Consecutive Sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> numSet = new(nums);

        int longestSequence = 0;

        foreach(int num in numSet) {
            int parentSequence = num - 1;
            if(!numSet.Contains(parentSequence)) {
                // if the value is the beginning of a sequence
                int length = 1;
                while (numSet.Contains(num + length)) {
                    // while the next value in the sequence is there
                    ++length;
                }
                longestSequence = Math.Max(length, longestSequence);
            }
        }
        return longestSequence;
    }
}
